src/elm/type.coffee

```diff
@@ -182,21 +203,123 @@ module.exports.Convert = class Convert extends Expression
+      return null
+    else
+      canConvertToType(ToBoolean, @operand, ctx)
 
-module.exports.IntervalTypeSpecifier = class IntervalTypeSpecifier extends UnimplementedExpression
+module.exports.ConvertsToDate = class ConvertsToDate extends Expression
+  constructor: (json) ->
+    super
+    @operand = json.operand
 
-module.exports.ListTypeSpecifier = class ListTypeSpecifier extends UnimplementedExpression
+  exec: (ctx) ->
+    operatorValue = @execArgs(ctx)
+    if operatorValue is null
+      return null
+    else
+      canConvertToType(ToDate, @operand, ctx)
 
-module.exports.NamedTypeSpecifier = class NamedTypeSpecifier extends UnimplementedExpression
+module.exports.ConvertsToDateTime = class ConvertsToDateTime extends Expression
+  constructor: (json) ->
+    super
+    @operand = json.operand
 
+  exec: (ctx) ->
+    operatorValue = @execArgs(ctx)
+    if operatorValue is null
+      return null
+    else
+      canConvertToType(ToDateTime, @operand, ctx)
+
+module.exports.ConvertsToDecimal = class ConvertsToDecimal extends Expression
+  constructor: (json) ->
+    super
+    @operand = json.operand
+
+  exec: (ctx) ->
+    operatorValue = @execArgs(ctx)
+    if operatorValue is null
+      return null
+    else
+      canConvertToType(ToDecimal, @operand, ctx)
+
+module.exports.ConvertsToInteger = class ConvertsToInteger extends Expression
+  constructor: (json) ->
+    super
+    @operand = json.operand
+
+  exec: (ctx) ->
+    operatorValue = @execArgs(ctx)
+    if operatorValue is null
+      return null
+    else
+      canConvertToType(ToInteger, @operand, ctx)
+
+module.exports.ConvertsToQuantity = class ConvertsToQuantity extends Expression
+  constructor: (json) ->
+    super
+    @operand = json.operand
+
+  exec: (ctx) ->
+    operatorValue = @execArgs(ctx)
+    if operatorValue is null
+      return null
+    else
+      canConvertToType(ToQuantity, @operand, ctx)
+
+module.exports.ConvertsToRatio = class ConvertsToRatio extends Expression
+  constructor: (json) ->
+    super
+    @operand = json.operand
+
+  exec: (ctx) ->
+    operatorValue = @execArgs(ctx)
+    if operatorValue is null
+      return null
+    else
+      canConvertToType(ToRatio, @operand, ctx)
+
+module.exports.ConvertsToString = class ConvertsToString extends Expression
+  constructor: (json) ->
+    super
+    @operand = json.operand
+
+  exec: (ctx) ->
+    operatorValue = @execArgs(ctx)
+    if operatorValue is null
+      return null
+    else
+      canConvertToType(ToString, @operand, ctx)
+
+module.exports.ConvertsToTime = class ConvertsToTime extends Expression
+  constructor: (json) ->
+    super
+    @operand = json.operand
+
+  exec: (ctx) ->
+    operatorValue = @execArgs(ctx)
+    if operatorValue is null
+      return null
+    else
+      canConvertToType(ToTime, @operand, ctx)
+
+canConvertToType = (toFunction, operand, ctx) ->
+  try
+    value = new toFunction({"type": toFunction.name, "operand": operand}).execute(ctx)
```

Comment:
Oh man, `new toFunction(...)` totally freaks me out.  I think it **is** the right way to do this, but still...